reactor-netty-core/src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -230,6 +231,108 @@ private String getPasswordValue() {
 			AdvancedByteBufFormat.HEX_DUMP
 					.toLoggingHandler("reactor.netty.proxy", LogLevel.DEBUG, Charset.defaultCharset());
 
+	static final String HTTP_PROXY_HOST = "http.proxyHost";
+	static final String HTTP_PROXY_PORT = "http.proxyPort";
+	static final String HTTPS_PROXY_HOST = "https.proxyHost";
+	static final String HTTPS_PROXY_PORT = "https.proxyPort";
+	static final String HTTP_NON_PROXY_HOSTS = "http.nonProxyHosts";
+	static final String DEFAULT_NON_PROXY_HOSTS = "localhost|127.*|[::1]";
+
+	static final String SOCKS_PROXY_HOST = "socksProxyHost";
+	static final String SOCKS_PROXY_PORT = "socksProxyPort";
+	static final String SOCKS_VERSION = "socksProxyVersion";
+	static final String SOCKS_VERSION_5 = "5";
+	static final String SOCKS_VERSION_4 = "4";
+	static final String SOCKS_USERNAME = "java.net.socks.username";
+	static final String SOCKS_PASSWORD = "java.net.socks.password";
+
+	@Nullable
+	static ProxyProvider createFrom(Properties properties) {
+		Objects.requireNonNull(properties, "properties");
+
+		if (properties.containsKey(HTTP_PROXY_HOST) || properties.containsKey(HTTPS_PROXY_HOST)) {
+			return createHttpProxyFrom(properties);
+		}
+		if (properties.containsKey(SOCKS_PROXY_HOST)) {
+			return createSocksProxyFrom(properties);
+		}
+
+		return null;
+	}
+
+	/*
+		assumes properties has either http.proxyHost or https.proxyHost
+	 */
+	static ProxyProvider createHttpProxyFrom(Properties properties) {
+		String hostProperty;
+		String portProperty;
+		String defaultPort;
+		if (properties.containsKey(HTTPS_PROXY_HOST)) {
+			hostProperty = HTTPS_PROXY_HOST;
+			portProperty = HTTPS_PROXY_PORT;
+			defaultPort = "443";
+		}
+		else {
+			hostProperty = HTTP_PROXY_HOST;
+			portProperty = HTTP_PROXY_PORT;
+			defaultPort = "80";
+		}
+
+		String hostname = Objects.requireNonNull(properties.getProperty(hostProperty), hostProperty);
+		int port = parsePort(properties.getProperty(portProperty, defaultPort), portProperty);
+
+		String nonProxyHosts = properties.getProperty(HTTP_NON_PROXY_HOSTS, DEFAULT_NON_PROXY_HOSTS);
+
+		return ProxyProvider.builder()
+				.type(ProxyProvider.Proxy.HTTP)
+				.host(hostname)
+				.port(port)
+				.nonProxyHosts(nonProxyHosts)
+				.build();
+	}
+
+	static ProxyProvider createSocksProxyFrom(Properties properties) {
+		String hostname = properties.getProperty(SOCKS_PROXY_HOST);
```

Comment:
shouldn't this be checked for null, same as `hostname` in `createHttpProxyFrom` method above?

Reply:
not really, at least for now. That method is called when socks proxy host is set.
Anyway, added a null check in this method too just in case.